spinoso-string/src/enc/mod.rs

```diff
@@ -147,6 +148,16 @@ impl EncodedString {
             EncodedString::Utf8(_) => Encoding::Utf8,
         }
     }
+
+    #[inline]
+    pub fn set_encoding(&mut self, encoding: Encoding) {
+        if self.encoding() == encoding {
+            return;
+        };
+        let s = mem::take(self);
+        let buf = s.into_buf();
+        *self = Self::new(buf, encoding);
```

Comment:
Nice!

Just checking, `mem::take` is letting this function own self, and then we're just setting that ownership back to self with the `*self = ` assignment right? That's very handy in this niche cache.

Reply:
`mem::take` moves the `EncodedString` struct out of `&mut self` and leaves `EncodedString::default()` in its place.

Then we grab its buf, construct a new `EncodedString` with the buf and given encoding.

Then we move that new `EncodedString` back into the `&mut self`.

throughout this whole dance, `self` is valid and initialized.